tensorboard/webapp/metrics/store/metrics_reducers.ts

```diff
@@ -933,6 +936,19 @@ const reducer = createReducer(
       },
     };
   }),
+  on(actions.metricsEnableSavingPinsToggled, (state) => {
+    const nextSavingPinsEnabled = !(
+      state.settingOverrides.savingPinsEnabled ??
+      state.settings.savingPinsEnabled
```

Comment:
I'm curious, did you find out the "[legacy reasons](https://github.com/tensorflow/tensorboard/blob/370132cd808ab64baaeba5366af12b247833f4a4/tensorboard/webapp/metrics/store/metrics_types.ts#L258-L259)" why we need settings as well as settingsOverrides?

Reply:
I may be mistaken, but based on my understanding, the  `settings` property is used for setting default values and loading persistent settings, while the  property `settingsOverrides` is used when a user makes changes, such as toggling a checkbox.

In the [code](https://github.com/tensorflow/tensorboard/blob/master/tensorboard/webapp/metrics/store/metrics_selectors.ts#L310-L318), the  `settingsOverrides` is used to override the original setting store value. I'm not sure why these properties are separated, but that's what I discovered while examining the code.

Reply:
Thanks, that makes it clearer!